pkg/volume/csi/csi_attacher.go

```diff
@@ -293,11 +276,11 @@ func (c *csiAttacher) MountDevice(spec *volume.Spec, devicePath string, deviceMo
 
 	// Store volume metadata for UnmountDevice. Keep it around even if the
 	// driver does not support NodeStage, UnmountDevice still needs it.
-	if err = os.MkdirAll(deviceMountPath, 0750); err != nil && !corruptedDir {
-		return errors.New(log("attacher.MountDevice failed to create dir %#v:  %v", deviceMountPath, err))
-	}
-	klog.V(4).Info(log("created target path successfully [%s]", deviceMountPath))
 	dataDir := filepath.Dir(deviceMountPath)
+	if err = os.MkdirAll(dataDir, 0750); err != nil {
```

Comment:
`deviceMountPath` is used as `NodeStageVolumeRequest.staging_target_path`. CSI says:

>  // The CO SHALL ensure
  // that the path is directory and that the process serving the
  // request has `read` and `write` permission to that directory. The
  // CO SHALL be responsible for creating the directory if it does not
  // exist.
  string staging_target_path = 3;

So, Kubernetes should `MkDir` whole `deviceMountPath`. Or CSI needs to be fixed. Or worked around, as we do with block volumes - CSI drivers stage block devices as *files* inside `staging_target_path` *directory*).

Reply:
are these addressed yet?